scripts/skills/actives/rf_pummel_skill.nut

```diff
@@ -0,0 +1,129 @@
+this.rf_pummel_skill <- ::inherit("scripts/skills/actives/line_breaker", {
+	m = {},
+	function create()
+	{
+		this.line_breaker.create();
+		this.m.ID = "actives.rf_pummel";
+		this.m.Name = "Pummel";
+		this.m.Description = "Use a heavy attack to knock back your target and take their place, all in one action."
+		this.m.Icon = "skills/rf_pummel_skill.png";
+		this.m.IconDisabled = "skills/rf_pummel_skill_sw.png";
+		this.m.Overlay = "rf_pummel_skill";
+		this.m.SoundOnUse = [
+			"sounds/combat/indomitable_01.wav",
+			"sounds/combat/indomitable_02.wav"
+		];
+		this.m.Order = ::Const.SkillOrder.Any;
+		this.m.ActionPointCost = 6;
+		this.m.FatigueCost = 20;
+		this.m.AIBehaviorID = ::Const.AI.Behavior.ID.LineBreaker;
+	}
+
+	function getTooltip()
+	{
+		local ret = this.skill.getDefaultUtilityTooltip();
+
+		ret.push({
+			id = 10,
+			type = "text",
+			icon = "ui/icons/special.png",
+			text = ::Reforged.Mod.Tooltips.parseString("Has a " + ::MSU.Text.colorGreen("100%") + " chance to [stagger|Skill+staggered_effect] the target")
+		});
+
+		ret.push({
+			id = 11,
+			type = "text",
+			icon = "ui/icons/special.png",
+			text = "If the attack is successful, the target is pushed back a tile and you move into their position"
+		});
+
+		local attack = this.getValidAttack();
+		if (attack == null)
+		{
+			ret.push({
+				id = 12,
+				type = "text",
+				icon = "ui/icons/warning.png",
+				text = ::MSU.Text.colorRed("Requires an attack from a two-handed hammer")
+			});
+		}
+		else
+		{
+			ret.push({
+				id = 12,
+				type = "text",
+				icon = "ui/icons/special.png",
+				text = ::Reforged.Mod.Tooltips.parseString(format("Perform a free [%s|Skill+%s] on the target", attack.getName(), split(::IO.scriptFilenameByHash(attack.ClassNameHash), "/").top()))
+			});
+		}
+
+		if (this.getContainer().getActor().getCurrentProperties().IsRooted)
+		{
+			ret.push({
+				id = 13,
+				type = "text",
+				icon = "ui/icons/warning.png",
+				text = ::MSU.Text.colorRed("Cannot be used while Rooted")
+			});
+		}
+
+		return ret;
+	}
+
+	function isUsable()
+	{
+		return this.line_breaker.isUsable() && !this.getContainer().getActor().getCurrentProperties().IsRooted && this.getValidAttack() != null;
+	}
+
```

Comment:
You have no `onVerifyTarget` function here.
Just like in `isUsable` you should also consult `LineBreaker` whether it's usable on a specific target.

Right now you can use Pummel on rooted or knockbackimmune enemies. LineBreaker doesn't double-check for rooted so you are theoretically able to push those enemies around with Pummel, and that doesn't seem intended (if I read the vanilla code correctly).
KnockBackImmune enemies will stop LineBreaker, but you still wasted additional fatigue in that case.

Reply:
This is the vanilla `onVerifyTarget` for `line_breaker.nut`. Seems like it already checks for rooted and knockback immune targets.
```squirrel
function onVerifyTarget( _originTile, _targetTile )
{
	return this.skill.onVerifyTarget(_originTile, _targetTile) && !_targetTile.getEntity().getCurrentProperties().IsRooted && !_targetTile.getEntity().getCurrentProperties().IsImmuneToKnockBackAndGrab && this.findTileToKnockBackTo(_originTile, _targetTile) != null;
}
```

Reply:
Ah, I overlooked that Pummel inherits from linebreaker. Forget what I said then.